### ISSUE TITLE
Nice Code Stealing

### ISSUE THREAD

**Issue by GreemDev** [YilianSource/brackeys-bot#252]
Love how I come across an interesting bot only to find it's a rip off without credit.

https://github.com/Ultz/Volte
project: structured the same way
You have similar extensions. Some of which, the logic is identical, even if they're one-liners. The project structure is a near-perfect match to my bot. Which, if you couldn't tell, has been around longer than this repository has.

No, I'm not the rip off. Volte is my brain child and this project just cloned the repository and made changes. Sure, there may be a lot of changes, but have a look at [this](https://github.com/YilianSource/brackeys-bot/blob/master/src/Core/Models/Logging/ModerationLogEntry.cs), it's almost identical to [this](https://github.com/Ultz/Volte/blob/v3/src/Core/Models/EventArgs/ModActionEventArgs.cs). At the very least I'd like credit on the *repo*.

**Comment by YilianSource** [comment]
Hey there, so good to hear from you!

I'm going to give you a bit of backstory on the project, but you can also skip to the TL;DR at the bottom.

Disregarding v1 of the bot, which was written in a different language, v2 was written pretty rushed, since we needed an urgent replacement for v1, which practically went down overnight. As such, the code written was practically unmaintainable and "patched together" by software standards. Which is why, almost a year ago, we decided to rewrite the project, this time with more patience, time and research.

Discord.Net is a really powerful library for making Discord bots running on C#, and only upon researching other bots which run the same library did I figure that out. Which is why, this time, I took my time looking into the Discord.Net docs (also looking at [Qmmands](https://github.com/Quahu/Qmmands) which functions similarly), as well as various existing C# Discord bots on GitHub.

It's true, I looked over your project, studying best practices, but so did I look at 4 other projects. Your claim that "this project just cloned the repository and made changes" is hurtful and frankly a disgrace to me and my contributors, who have devoted time, patience and love into this project.

I am curious though, as to why you would publish your project on GitHub with a permissive MIT license if you wouldn't want other people to look at, study and learn from your project.

As to your claim ([ours](https://github.com/YilianSource/brackeys-bot/blob/master/src/Core/Models/Logging/ModerationLogEntry.cs) vs. [yours](https://github.com/Ultz/Volte/blob/v3/src/Core/Models/EventArgs/ModActionEventArgs.cs)) I am almost hurt that you accredit the invention of a builder-design-pattern to yourself. We don't even use `EventArgs`, and do you know the reason why they use the same properties? Because they do the same thing. Even the usage of similar extensions makes sense because they eliminate the same bothers that we would have without them.

**TL;DR:** No, we won't accredit your repository.

I know that this reply most likely will not be satisfactory to you, but I hope I was at least able to widen your understanding and appreciation of public sources and the luxury of research.

---

I also find it highly unprofessional to bring up these issues like this, and then either yourself or people in your name go and harass another server and attempt to publicly humiliate programmers. I would advise you, for future reference, to contact the developer of a repository by email or direct message to resolve any issues you may have with them, and try to talk it out rather than instantly blatantly insulting them with "nice code stealing".

**Comment by GreemDev** [comment]
I'm fine with people looking over my repository and taking inspiration from it. In fact, that's a benefit. But the thing is some files (as is with ModerationLogEntry) the thing that gets me is the fact that it's the little things that make it look copied. Now, not every file is an exact copy, which is a good thing, but you've never look at another project and been like "I've seen this before..."

________________________________

**Comment by GreemDev** [comment]
I'm fine with people looking over my repository and taking inspiration from it. In fact, that's a benefit. But the thing is some files (as is with ModerationLogEntry) the thing that gets me is the fact that it's the little things that make it look copied. Now, not every file is an exact copy, which is a good thing, but you've never look at another project and been like "I've seen this before..."

Also, what "other server" did I harass?

**Comment by YilianSource** [comment]
I totally understand what you mean. But just because we are doing the same thing doesn't mean that one copied from the other. If I may take the `ModerationLogEntry` as a repeated example, the shared fields are the things that are just necessary for a useable log entry, and the public methods are just as necessary to populate those fields.

Usually, if I see a file that is close to identical, I think to myself: "Hey, that's cool, that must mean that my idea isn't half bad!".

I'm referring to you (if that was you) entering the Discord server and insisting on the copyright violation, linking this issue, reflecting badly on us in general. If that didn't happen to be you, then I apologize.

**Comment by GreemDev** [comment]
Yeah, I know what ModerationLogEntry is for, but the gripe I have is that we used the builder pattern for the same part of the bot, and I don't have any other objects that used the builder pattern. I know the entire bot isn't a rip off anymore, and I respect you for that, but I still can't get over of how similar it's structured and the fact that some files (as mentioned above) contain similar conventions.

**Comment by GreemDev** [comment]
> I'm referring to you (if that was you) entering the Discord server and insisting on the copyright violation, linking this issue, reflecting badly on us in general. If that didn't happen to be you, then I apologize.

I was in the Discord as Greem#1337; I do believe someone named johnk from the organization I'm in did join your discord and do what you said.

**Comment by YilianSource** [comment]
We've used the builder object in another place, just because it's a convenient pattern to use for completing object information. The structure and conventions are coincidentally similar, as a result of restructuring and organization. In my eyes, this does not qualify for substantial or even partial code usage.

> I was in the Discord as Greem#1337; I do believe someone named johnk from the organization I'm in did join your discord and do what you said.

I see, then I do apologize for that claim, but I'd appreciate it if you'd be so kind as to forward this conversation to him. If he has additional thoughts he is free to add them, otherwise, I would feel that this is settled.

**Comment by GreemDev** [comment]
Agreed. But do you see the dilemma I had when I first discovered this repository? It was like deja vu.

**Comment by Perksey** [comment]
:wave: Head honcho from @Ultz here.

I'd like to apologise on behalf of @john-h-k, another one of the Ultz developers, for the inadverent ruckus he may have caused in the Discord server. I didn't know he was going to do that, and while I appreciate the defense of Volte's IP, this shouldn't have been done on such a mass public forum. Preferably all discussion of the matter should've been constrained to this GitHub issue.

When Greem made me aware of this I did look into the similarities between the codebase and, while the structure is pretty much the same, I don't believe that it is in violation of any copyright held by Greem or Ultz due to the divergence of this bot's codebase from the original. Some files more strongly matched than others, however the consensus was that while it _feels_ the same, it is derivative enough to not infringe on any of our IP.

I sincerely apologise for this whole mess on behalf of all parties involved, and any inconvenience caused by us. We should've looked into this further before making any rash judgements or actions. We're not going to pursue this issue further, however if you did want to link to the bots that inspired this bot in good faith (similar to how Volte has linked to Abyss for inspiring Volte's codebase) then please do! But we're not going to force or demand you to do so.

**Comment by YilianSource** [comment]
I really appreciate the apology, @Perksey, I'm glad we were able to resolve this! I'll close this issue soon then.

As to us linking your bot, I do admit that I am inclined to do so.
We're currently thinking up a new version that might even include a dashboard and SQL database, so not a lot of new progress will be made on this probably.

As a closer, I'd like to congratulate you and your team on your project and the code quality contained and we would be happy to help you in the future if you need anything.
